ssh-server/main.go

```diff
@@ -113,38 +118,99 @@ func serviceChannel(newChannel ssh.NewChannel) error {
 	}
 	defer channel.Close()
 
+	allow := map[string]bool{"shell": true, "pty-req": true, "window-change": true}
+	resize := make(chan *pty.Winsize, 1)
+	defer close(resize)
+	done := make(chan struct{})
+	for {
+		select {
+		case <-done:
+			return nil
+		case req, ok := <-requests:
+			if !ok {
+				return nil
+			}
+			if req.WantReply {
+				req.Reply(allow[req.Type], nil)
+			}
+			switch req.Type {
+			case "pty-req":
+				r := bytes.NewReader(req.Payload)
+				term, err := readString(r)
+				if err != nil {
+					return fmt.Errorf("parsing TERM environment variable value: %v", err)
+				}
+
+				size, err := readWindowSize(r)
+				if err != nil {
+					return fmt.Errorf("parsing window size: %v", err)
+				}
+				resize <- size
+
+				go func() {
+					if err := runPTY(channel, term, resize); err != nil {
+						fmt.Printf("Failed to run pty: %v", err)
```

Comment:
Instead of done being a channel with no information, consider making it `chan error` and then you can simply send the error here and return it from the function normally.

Reply:
good idea, thanks